src/services/upper-bar-manager/ui/dropdown-bar-item/dropdown-bar-item.tsx

```diff
@@ -0,0 +1,92 @@
+import { h, Fragment, VNode } from 'preact';
+import { useState, useRef, useLayoutEffect } from 'preact/hooks';
+import * as styles from './dropdown-bar-item.scss';
+import { ui } from '@playkit-js/kaltura-player-js';
+import { A11yWrapper } from '@playkit-js/common/dist/hoc/a11y-wrapper';
+import { SvgIcon } from '../../models/svg-icon';
+const { Icon, Tooltip } = ui.Components;
+
+type DropdownBarItemProps = {
+  displayName: string;
+  text: string;
+  icon: SvgIcon;
+  onClick: (e: KeyboardEvent | MouseEvent) => void;
+  onDropdownClick: () => void;
+  tooltipPosition: string;
+};
+
+const PADDING = 11;
+
+const DropdownBarItem = ({ displayName, text, icon, onClick, onDropdownClick, tooltipPosition }: DropdownBarItemProps) => {
+  const comparisonTextRef = useRef<HTMLSpanElement | null>(null);
+  const textRef = useRef<HTMLSpanElement | null>(null);
+
+  const [showTooltip, setShowTooltip] = useState(false);
+  const [isFinalized, setIsFinalized] = useState(false);
```

Comment:
dont we already have this code somewhere ?

Reply:
yes, i took it from here https://github.com/kaltura/playkit-js-detect-ad-block/blob/master/src/components/text-with-tooltip/text-with-tooltip.tsx 
the idea is the same